test/alu_8bit_inc_test.c

```diff
@@ -0,0 +1,90 @@
+#include <criterion/criterion.h>
+#include <criterion/new/assert.h>
+#include <criterion/parameterized.h>
+
+#include "fixtures/cpu_mmu.h"
+#include "common/util.h"
+#include "common/alu.h"
+
+ParameterizedTestParameters(INC_n, INC_n) {
+    static TestParams params[] = {
+        // half carry
+        {0x04, 0, 15, offsetof(CPU, BC), offsetof(DoubleWordReg, words.hi), 16, 1, 0b00100000},
+        {0x0C, 0, 15, offsetof(CPU, BC), offsetof(DoubleWordReg, words.lo), 16, 1, 0b00100000},
+        {0x14, 0, 15, offsetof(CPU, DE), offsetof(DoubleWordReg, words.hi), 16, 1, 0b00100000},
+        {0x1C, 0, 15, offsetof(CPU, DE), offsetof(DoubleWordReg, words.lo), 16, 1, 0b00100000},
+        {0x24, 0, 15, offsetof(CPU, HL), offsetof(DoubleWordReg, words.hi), 16, 1, 0b00100000},
+        {0x2C, 0, 15, offsetof(CPU, HL), offsetof(DoubleWordReg, words.lo), 16, 1, 0b00100000},
+        {0x3C, 0, 15, offsetof(CPU, AF), offsetof(DoubleWordReg, words.hi), 16, 1, 0b00100000},
+    };
+
+    return cr_make_param_array(TestParams, params, sizeof(params) / sizeof(TestParams));
+}
+
+ParameterizedTest(TestParams *params, INC_n, INC_n, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    CPU_REG_F = 0;
```

Comment:
PR #57 added the clear_flag_register function to cpu.h which should be available after merging main